boa3/analyser/supportedstandard/standardanalyser.py

```diff
@@ -138,5 +139,46 @@ def _get_all_imports(self) -> List[importsymbol.Import]:
         return all_imports
 
     def _check_other_implemented_standards(self):
-        # TODO: Implement when detecting standards from implemented methods and events
-        pass
+        other_standards = supportedstandard.neo_standards.copy()
+        # verify only standards that were not mentioned
+        for standard in self.standards:
+            if standard in supportedstandard.neo_standards:
+                other_standards.pop(standard)
+
+        # gets a list of all events
+        events = [symbol for symbol in self.symbols.copy().values() if isinstance(symbol, Event)]
+        for imported in self._get_all_imports():
+            events.extend([event for event in imported.all_symbols.values() if isinstance(event, Event)])
+
+        # verify if the methods and events that were implemented corresponds to a standard
+        for standard in other_standards:
+
+            # verify the methods
+            methods_implemented = True
+            for standard_method in other_standards[standard].methods:
+                method_id = standard_method.external_name
+                found_methods = self.get_methods_by_display_name(method_id)
+
+                methods_implemented = any(
+                     other_standards[standard].match_definition(standard_method, method) for method in found_methods
+                )
+
+                if not methods_implemented:
+                    break
```

Comment:
since the `method_implemented` is also a validator for ending the loop, this would be better implemented as a `while` than a `for`

```suggestion
            methods_implemented = True
            index = 0
            standard_methods = other_standards[standard].methods
            
            while methods_implemented and index < len(standard_methods):
                standard_method = standard_methods[index]
                method_id = standard_method.external_name
                found_methods = self.get_methods_by_display_name(method_id)

                methods_implemented = any(
                     other_standards[standard].match_definition(standard_method, method) for method in found_methods
                )
                index += 1
                
```